File: 백준/Bronze/4153. 직각삼각형/직각삼각형.cc
#include <iostream>
using namespace std;

int main()
{
	int x, y, z;
	while (true) {
		cin >> x >> y >> z;
		if (x == 0 && y == 0 && z == 0) break;
		else {
			if (x > y) {
				if (x > z) {
					if (x * x == y * y + z * z) cout << "right" << endl;
					else cout << "wrong" << endl;
				}
				else {
					if (z * z == x * x + y * y) cout << "right" << endl;
					else cout << "wrong" << endl;
				}
			}
			else {
				if (y > z) {
					if (y * y == x * x + z * z) cout << "right" << endl;
					else cout << "wrong" << endl;
				}
				else {
					if (z * z == x * x + y * y) cout << "right" << endl;
					else cout << "wrong" << endl;
				}
			}
		}
	}
}

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);

        int a, b, c;

        while (true) {
            a = s.nextInt();
            b = s.nextInt();
            c = s.nextInt();

            if (a == 0 && b == 0 && c == 0) break;

            if (a > b) {
                if (a > c) { // a가 가장 클 때
                    if (Math.pow(b, 2) + Math.pow(c, 2) == Math.pow(a, 2)) {
                        System.out.println("right");
                    } else {
                        System.out.println("wrong");
                    }
                } else { // c가 가장 클 때
                    if (Math.pow(a, 2) + Math.pow(b, 2) == Math.pow(c, 2)) {
                        System.out.println("right");
                    } else {
                        System.out.println("wrong");
                    }
                }
            } else {
                if (b > c) { // b가 가장 클 때
                    if (Math.pow(a, 2) + Math.pow(c, 2) == Math.pow(b, 2)) {
                        System.out.println("right");
                    } else {
                        System.out.println("wrong");
                    }
                } else { // c가 가장 클 때
                    if (Math.pow(a, 2) + Math.pow(b, 2) == Math.pow(c, 2)) {
                        System.out.println("right");
                    } else {
                        System.out.println("wrong");
                    }
                }
            }
        }

        s.close();
    }
}